### DISCUSSION TITLE
How to "loop mode"

### DISCUSSION THREAD

**Discussion by kirgans** [TanStack/virtual#600] [Q&A]
Quite simple question but how can I get a "loop mode", when I get to the last index, loop to the first one ?

Thanks :)

**Comment by piecyk** [comment]
This is not that simple question, one idea could be having a list of 100 items, you need to render extra items before and after [97, 98, 99, 0, 1, 2, ..., 99, 0, 1, 2, 3] and when scrolling down hitting the last element need to scroll to first, and when scrolling up hitting first need to scroll to last.

**Reply by kirgans** [reply]
I think there is something to tweak with the overscan, but this would definitely be a great idea of a **new feature** !

**Reply by piecyk** [reply]
Yes, you can use overscan with `rangeExtractor` to return indexes like [97, 98, 99, 0, 1, 2, ..., 99, 0, 1, 2, 3]

**Reply by kirgans** [reply]
Can you provide an example on how to do it please ? I can't find example for rangeExtractor...

**Reply by kirgans** [reply]
@piecyk  And how can I do to not scroll back to the first item, but keep scrolling in the same direction and add the first item to the end of the array ?

**Comment by romankoho** [comment]
@spacedev75 were you able to solve this? I'd also need to implement a loop in my virtualized list.
